src/account-type-utils.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * Returns the account type based on the user's email address
+ * to use as a Mixpanel property on the user profile.
+ *
+ */
+
+const generalTld = [
+  'com', 'net', 'io', 'ai', 'co', 'app', 'us', 'uk', 'ca',
+  'de', 'fr', 'jp', 'au', 'ru', 'ch', 'se', 'no', 'nl', 'it',
+  'es', 'dk', 'cz', 'br', 'be', 'at', 'ar', 'in', 'mx', 'pl',
+  'pt', 'fi', 'gr', 'hk', 'id', 'ie', 'il', 'is', 'kr', 'my',
+  'nz', 'ph', 'sg', 'th', 'tw', 'vn', 'cn'
+]
+
+const domainTlds = {
+  'broadinstitute.org': 'Broad Employee',
+  'firecloud.org': 'Broad Employee',
+  'verily.com': 'Verily Employee',
+  'google.com': 'Verily Employee',
+  'gmail.com': 'Independent User',
+  'gserviceaccount.com': 'Service Account User',
+  'edu': 'Educational Institute User',
+  'org': 'Non-profit User',
+  'gov': 'Government User'
+}
+
+const validateEmail = email => {
+  return String(email)
+    .toLowerCase()
+    .match(
+      /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+    )
+}
+
+const getAccountType = email => {
+  if (!validateEmail(email)) {
+    return 'Other'
+  }
+
+  const emailDomain = email.split('@').pop()
+
+  for (const [domain, accountType] of Object.entries(domainTlds)) {
+    if (emailDomain.endsWith(domain) || generalTld.some(tld => emailDomain.endsWith(`.${domain}.${tld}`))) {
+      return accountType
+    }
+  }
```

Comment:
```suggestion
  const accountType = _.find(baseTld => {
    const allTlds = [`.${baseTld}, ..._.map(tld => `.${baseTld}.${tld}`, generalTld)]
    _.includes(emailDomain, allTlds)
  }, domainTlds)
```
I didn't test this code, but something like this is in the more functional style that we prefer. Also adds `.` to the beginning of each of the domains (if they're not added to the keys of `domainTlds`).

I'm pretty sure this will do effectively the same thing as the code you have, though without the `return`, so you'll have to reconcile that with the `if` block below (that I haven't fully understood yet).

Reply:
Fixed that